src/main/java/seedu/address/ui/DeadlineTaskListPane.java

```diff
@@ -1,26 +1,71 @@
 package seedu.address.ui;
 
+import java.time.LocalDateTime;
 import java.util.List;
+import java.util.logging.Logger;
 
+import javafx.beans.binding.Bindings;
+import javafx.collections.ListChangeListener;
 import javafx.collections.ObservableList;
 import javafx.fxml.FXML;
+import javafx.scene.control.Label;
 import javafx.scene.control.ListCell;
 import javafx.scene.control.ListView;
 import javafx.scene.layout.Pane;
+import seedu.address.commons.core.LogsCenter;
 import seedu.address.model.IndexedItem;
+import seedu.address.model.ReadOnlyModel;
+import seedu.address.model.filter.TaskOverduePredicate;
+import seedu.address.model.filter.TaskUnfinishedPredicate;
+import seedu.address.model.filter.TaskWillHappenTodayPredicate;
 import seedu.address.model.task.DeadlineTask;
 
 public class DeadlineTaskListPane extends UiPart<Pane> {
 
     private static final String FXML = "/view/DeadlineTaskListPane.fxml";
 
+    private static final Logger logger = LogsCenter.getLogger(DeadlineTaskListPane.class);
+
+    private final ObservableList<IndexedItem<DeadlineTask>> listedDeadlineTasks;
+    private ObservableList<DeadlineTask> todayDeadlineTasks;
+    private ObservableList<DeadlineTask> overdueDeadlineTasks;
+    private ObservableList<DeadlineTask> unfinishedDeadlineTasks;
+
     @FXML
     private ListView<IndexedItem<DeadlineTask>> deadlineTaskListView;
 
-    public DeadlineTaskListPane(ObservableList<IndexedItem<DeadlineTask>> deadlineTaskList) {
+    @FXML
+    private Label listedDeadlineCounter;
+    @FXML
+    private Label todayUnfinishedDeadlineCounter;
+    @FXML
+    private Label overdueDeadlineCounter;
+    @FXML
+    private Label unfinishedDeadlineCounter;
+
+    public DeadlineTaskListPane(ReadOnlyModel model) {
         super(FXML);
-        deadlineTaskListView.setItems(deadlineTaskList);
+
+        // Initialize task lists
+        this.listedDeadlineTasks = model.getDeadlineTaskList();
+        this.todayDeadlineTasks = model.getDeadlineTaskList(new TaskWillHappenTodayPredicate(LocalDateTime.now()));
```

Comment:
@nightism Should these counters be limited to what is actually shown in the UI or not? As of now, your `listedDeadlineTasks` is showing the filtered list, while `todayDeadlineTasks`, `overdueDeadlineTasks` is showing the unfiltered list!!!!